src/laser/MultiLaser.cpp

```diff
@@ -243,44 +257,80 @@ MultiLaser::GetEnvelopeFromFile (const amrex::Box& domain) {
     AMREX_ALWAYS_ASSERT(axis_labels[0] == "t" && axis_labels[1] == "y" && axis_labels[2] == "x");
 
     const std::shared_ptr<input_type> data = laser_comp.loadChunk<input_type>();
-
-    auto extend = laser_comp.getExtent();
-
+    auto extent = laser_comp.getExtent();
     double unitSI = laser_comp.unitSI();
 
-    //hipace: xyt in Fortran order
-    //lasy: tyx in C order
+    // Extract grid offset and grid spacing from laser file
+    std::vector<double> offset = laser.gridGlobalOffset();
+    std::vector<double> position = laser_comp.position<double>();
+    std::vector<double> spacing = laser.gridSpacing<double>();
 
-    if (static_cast<std::uint64_t>(domain.length(0)) != extend[2] ||
-        static_cast<std::uint64_t>(domain.length(1)) != extend[1] ||
-        static_cast<std::uint64_t>(domain.length(2)) != extend[0]) {
-        amrex::Abort("Incompatible box sizes. HiPACE++ (" +
-        std::to_string(domain.length(0)) + ", " + std::to_string(domain.length(1)) + ", "
-        + std::to_string(domain.length(2)) + "), "
-        + "File (" + std::to_string(extend[2]) + ", " + std::to_string(extend[1]) + ", "
-        + std::to_string(extend[0]) + ")\n");
-    }
+    // Calculate the min and max of the grid from laser file
+    amrex::Real ymin_laser = offset[1] + position[1]*spacing[1];
+    amrex::Real xmin_laser = offset[2] + position[2]*spacing[2];
+    AMREX_ALWAYS_ASSERT(position[0] == 0 && position[1] == 0 && position[2] == 0);
```

Comment:
The code was not tested with a non-zero value. We're not sure if it will be useful or not, so for the moment we keep it and the first time a non-zero value is encountered, we'll hit this assert and do the tests so we can remove it safely. In the long run, if this is never non-zero, we'll simplify the code (and keep the assert).